### ISSUE TITLE
Merged mining with Antminer failed, more investigation needed

### ISSUE THREAD

**Issue by jansegre** [HathorNetwork/hathor-core#11]

**Comment by jansegre** [comment]
This blocks #3

**Comment by jansegre** [comment]
After some investigation it appears that we're calculating the difficulty incorrectly.

In practice mining should still work though most shares will be rejected. The result of those shares is actually used though and it stills contributes to the network.

But this is definitely not how it should work. I'm already testing a fix and maybe I'll manage to update the network today.

**Comment by jansegre** [comment]
Just some heads up: after the difficulty calculation was fixed there were more tests but other problems arose. There was a stratum extension (`mininig.configure`) the server did not implement and caused the Antminer not to connect. The initial implementation that naively answer that no configuration could be set (instead of method not found) only briefly solved the problem. I suspected that a particular configuration (minimum difficulty) was the cause of the latest concern, that configuration has been implemented but tests have yet to be made.
